tests/app/shared/issue/label/label.component.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { LabelComponent } from '../../../../../src/app/shared/issue/label/label.component';
+import { LabelService } from '../../../../../src/app/core/services/label.service';
+import { PermissionService } from '../../../../../src/app/core/services/permission.service';
+import { ISSUE_WITH_EMPTY_DESCRIPTION } from '../../../../constants/githubissue.constants';
+import { Issue } from '../../../../../src/app/core/models/issue.model';
+import { SEVERITY_LABELS, COLOR_SEVERITY_LOW, SEVERITY, COLOR_SEVERITY_HIGH, SEVERITY_HIGH, SEVERITY_MEDIUM } from '../../../../constants/label.constants';
+import { of } from 'rxjs';
+
+describe('LabelComponent', () => {
+  let labelComponent: LabelComponent;
+  let issueService: any;
+  let permissionService: any;
+  let labelService: any;
+  let thisIssue: Issue;
+
+  const formBuilder: any = null;
+  const errorHandlingService: any = null;
+
+  beforeEach(() => {
+    labelService = jasmine.createSpyObj(LabelService, ['getLabelList',
+      'getColorOfLabel', 'isDarkColor', 'getRequiredLabelsAsArray']);
+    issueService = jasmine.createSpyObj('IssueService', ['updateIssue']);
+    permissionService = new PermissionService(null, null, null);
+
+    labelComponent = new LabelComponent(issueService, formBuilder, errorHandlingService, labelService, permissionService);
+    thisIssue =  Issue.createPhaseBugReportingIssue(ISSUE_WITH_EMPTY_DESCRIPTION);
+    labelComponent.issue = thisIssue;
+    labelComponent.attributeName = SEVERITY;
+  });
+
+  it('should be initialised with a list of label values and a labelColor', () => {
+    labelService.getLabelList.and.returnValue(SEVERITY_LABELS);
+    labelService.getColorOfLabel.and.returnValue(COLOR_SEVERITY_LOW);
+    labelComponent.ngOnInit();
+    labelComponent.ngOnChanges();
+
+    expect(labelComponent.labelValues).toEqual(SEVERITY_LABELS);
+    expect(labelComponent.labelColor).toEqual(COLOR_SEVERITY_LOW);
+  });
+
+  it('should change label color when updateLabel is called', () => {
+    labelService.getLabelList.and.returnValue(SEVERITY_LABELS);
+    labelService.getColorOfLabel.and.returnValue(COLOR_SEVERITY_LOW);
+    labelComponent.ngOnInit();
+    labelComponent.ngOnChanges();
+
+    const latestIssue = {
+        severity: SEVERITY_HIGH,
+        ...ISSUE_WITH_EMPTY_DESCRIPTION
+    };
+    labelService.getColorOfLabel.and.returnValue(COLOR_SEVERITY_HIGH);
+    issueService.updateIssue.and.returnValue(of(latestIssue));
+    labelComponent.updateLabel(SEVERITY_HIGH);
+
+    expect(labelComponent.labelValues).toEqual(SEVERITY_LABELS);
+    expect(labelComponent.labelColor).toEqual(COLOR_SEVERITY_HIGH);
+  });
```

Comment:
Should we also create a spy for the `issueUpdated` EventEmitter held by the LabelComponent ?
We can verify that the `emit` method is called on this EventEmitter, using something like:
`expect(issueUpdated.emit.calls.count()).toEqual(1);`

Here's a reference for Jasmine spies: [link](https://daveceddia.com/jasmine-2-spy-cheat-sheet/)

Reply:
Alright 👍